test/lib/test-helper.ts

```diff
@@ -6,59 +6,101 @@ import Promise = require('bluebird');
 import io = require('socket.io-client');
 
 var log = debug('child_server:stdout');
+// CONST
+var SESSION_EXPIRATION: number = 3;
+var MAXBUFFERSIZE: number = 3;
+var POLL_TIMEOUT: number = 1000;
+
+function createCertificate(): Promise<any>
+{
+    return (Promise.promisify(child.execFile))('./scripts/gen_keys.sh', ['./temp']);
+}
+
+function removeCertificate(): Promise<any>
+{
+    return (Promise.promisify(child.exec))('rm -rf ./temp');
+}
 
 export interface IOurSocket extends SocketIOClient.Socket {
     emitAcknowledge(event: string, arg?: any): Promise<void>;
 }
 
+export interface IOurChildProcess extends child.ChildProcess {
+    isHTTPS: boolean;
+}
+
 type ReturnValue = {
-    server: child.ChildProcess;
+    server: IOurChildProcess;
     ipc: IOurSocket;
 }
 
-export function startServer(testport: number = 3333): Promise<ReturnValue>
+export function startServer(https: boolean = false, testport: number = 3333): Promise<ReturnValue>
 {
-    var server = child.spawn('node',
-                             [
-                                './test/lib/test-index.js',
-                                '--port=3000',
-                                '--testport=' + testport
-                             ],
-                             {
-                                stdio: [null, null, null, 'ipc']
-                             }
-                            );
+    var serverOptions = [
+        './test/lib/test-index.js',
+        '--port=3000',
+        '--testport=' + testport,
+        '--logging-stdoutLevel=debug'
+    ];
+    if (https) {
+        serverOptions.push('--https-enable=true');
+        serverOptions.push('--https-ca=../temp/ca-cert.pem');
+        serverOptions.push('--https-cert=../temp/server-cert.pem');
+        serverOptions.push('--https-key=../temp/server-key.pem');
+        serverOptions.push('--logging-clientDetail=true');
+        serverOptions.push('--session-expiration=' + SESSION_EXPIRATION);
+        serverOptions.push('--longpoll-maxbuffersize=' + MAXBUFFERSIZE);
+        serverOptions.push('--longpoll-polltimeout=' + POLL_TIMEOUT);
+    }
+    var p: Promise<any> = https ? createCertificate() : Promise.resolve();
```

Comment:
I would make this synchronous if possible, it just simplifies everything and it is just for testing.